lib/streamlit/connections/snowflake_connection.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: We won't always be able to import from snowflake.{connector, snowpark}.* so need
+# the `type: ignore` comment below, but that comment will explode if `warn-unused-ignores`
+# is turned on when the package is available. Unfortunately, mypy doesn't provide a good
+# way to configure this at a per-line level :(
+# mypy: no-warn-unused-ignores
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import TYPE_CHECKING, Any, Dict, Sequence, cast
```

Comment:
You're using future annotations, instead of `typing.Dict` you should just use `dict`
```suggestion
from typing import TYPE_CHECKING, Any, Sequence, cast
```

Reply:
Pedant Trigger Alert 😆 

Future annotations don't directly permit the use of `dict` instead of `typing.Dict` but it has that effect because PEP 563 [defers annotation evaluation](https://peps.python.org/pep-0563/) at function definition time, and instead stores them as strings.  `typing.Dict` is deprecated [as of Python 3.9](https://docs.python.org/3/library/typing.html#typing.Dict) in favor of `dict`, but I think Python 3.8 is probably still your minimum version.

So Mark is correct; since you're using future annotations, you might as well save yourself an import.  And "lucky" for us, [PEP 649](https://peps.python.org/pep-0649/) will take over the world as of Python 3.13 and beyond.